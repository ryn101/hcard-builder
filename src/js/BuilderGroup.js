import Unique from './Unique';
import BuilderGroupDetail from './BuilderGroupDetail';

/**
 * Class for storing builder group specific information
 */
class BuilderGroup extends Unique {
    constructor(title, modelName = null, details = []) {
        super('bg');
        this.title = title;
        this.modelName = modelName;
        this.details = details;
    }

    addDetail(description, modelName) {
        this.details.push(new BuilderGroupDetail(description, modelName));
    }
}

export default BuilderGroup;
