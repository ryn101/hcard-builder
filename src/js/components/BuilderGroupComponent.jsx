import React from 'react';
import PropTypes from 'prop-types';
import BuilderGroup from '../BuilderGroup';
import BuilderGroupDetailComponent from './BuilderGroupDetailComponent';

function BuilderGroupComponent(props) {
    const { builderGroup, onDetailUpdated, modelValue } = props;

    return (
        <section className="builder-group">
            <h2>{builderGroup.title}</h2>
            <ul className="flex-row flex-row--columns-2">
                {
                    builderGroup.details.map((bgd) => (
                        <BuilderGroupDetailComponent
                            key={bgd.id}
                            builderGroup={builderGroup}
                            builderGroupDetail={bgd}
                            modelValue={modelValue}
                            onDetailUpdated={onDetailUpdated}
                        />
                    ))
                }
            </ul>
        </section>
    );
}

BuilderGroupComponent.propTypes = {
    builderGroup: PropTypes.instanceOf(BuilderGroup).isRequired,
    onDetailUpdated: PropTypes.func,
    // eslint-disable-next-line react/forbid-prop-types
    modelValue: PropTypes.object,
};

BuilderGroupComponent.defaultProps = {
    onDetailUpdated: () => {},
    modelValue: null,
};

export default BuilderGroupComponent;
