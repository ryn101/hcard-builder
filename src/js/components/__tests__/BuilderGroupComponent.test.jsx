import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BuilderGroupComponent from '../BuilderGroupComponent';
import BuilderGroup from '../../BuilderGroup';

const builderGroup = new BuilderGroup('Test');
builderGroup.addDetail('Test Detail', 'testDetail');
builderGroup.addDetail('Another Detail', 'anotherDetail');

configure({ adapter: new Adapter() });

it('renders the builder group without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BuilderGroupComponent builderGroup={builderGroup} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('once the builder group has rendered', () => {
    let builderGroupNode;

    beforeEach(() => {
        builderGroupNode = shallow(<BuilderGroupComponent builderGroup={builderGroup} />);
    });

    it('renders with state correctly', () => {
        expect(builderGroupNode).toMatchSnapshot();
    });

    it('renders the title correctly', () => {
        const titleNode = builderGroupNode.find('h2');
        expect(titleNode.text()).toEqual(builderGroup.title);
    });
});
