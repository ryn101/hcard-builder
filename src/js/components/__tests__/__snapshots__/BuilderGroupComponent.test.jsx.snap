// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`once the builder group has rendered renders with state correctly 1`] = `
<section
  className="builder-group"
>
  <h2>
    Test
  </h2>
  <ul
    className="flex-row flex-row--columns-2"
  >
    <BuilderGroupDetailComponent
      builderGroup={
        BuilderGroup {
          "details": Array [
            BuilderGroupDetail {
              "description": "Test Detail",
              "id": "bgd-1",
              "modelName": "testDetail",
            },
            BuilderGroupDetail {
              "description": "Another Detail",
              "id": "bgd-2",
              "modelName": "anotherDetail",
            },
          ],
          "id": "bg-1",
          "modelName": null,
          "title": "Test",
        }
      }
      builderGroupDetail={
        BuilderGroupDetail {
          "description": "Test Detail",
          "id": "bgd-1",
          "modelName": "testDetail",
        }
      }
      key="bgd-1"
      modelValue={null}
      onDetailUpdated={[Function]}
    />
    <BuilderGroupDetailComponent
      builderGroup={
        BuilderGroup {
          "details": Array [
            BuilderGroupDetail {
              "description": "Test Detail",
              "id": "bgd-1",
              "modelName": "testDetail",
            },
            BuilderGroupDetail {
              "description": "Another Detail",
              "id": "bgd-2",
              "modelName": "anotherDetail",
            },
          ],
          "id": "bg-1",
          "modelName": null,
          "title": "Test",
        }
      }
      builderGroupDetail={
        BuilderGroupDetail {
          "description": "Another Detail",
          "id": "bgd-2",
          "modelName": "anotherDetail",
        }
      }
      key="bgd-2"
      modelValue={null}
      onDetailUpdated={[Function]}
    />
  </ul>
</section>
`;
